File: flutter_app/lib/core/models/imagen.dart
class Imagen {
  final String id;
  final String path;
  final String thumbPath;
  final bool esPrincipal;
  final double? areaX;
  final double? areaY;
  final double? areaW;
  final double? areaH;

  Imagen({
    required this.id,
    required this.path,
    required this.thumbPath,
    this.esPrincipal = false,
    this.areaX,
    this.areaY,
    this.areaW,
    this.areaH,
  });

  factory Imagen.fromJson(Map<String, dynamic> json) => Imagen(
        id: json['id'] as String,
        path: json['path'] as String,
        thumbPath: json['thumb_path'] as String,
        esPrincipal: json['es_principal'] as bool? ?? false,
        areaX: (json['area_x'] as num?)?.toDouble(),
        areaY: (json['area_y'] as num?)?.toDouble(),
        areaW: (json['area_w'] as num?)?.toDouble(),
        areaH: (json['area_h'] as num?)?.toDouble(),
      );
}

File: go_backend/models/imagen.go
package models

type Imagen struct {
	ID          string   `json:"id"`
	ObjetoID    string   `json:"objeto_id"`
	Path        string   `json:"path"`
	ThumbPath   string   `json:"thumb_path"`
	AreaX       *float64 `json:"area_x,omitempty"`
	AreaY       *float64 `json:"area_y,omitempty"`
	AreaW       *float64 `json:"area_w,omitempty"`
	AreaH       *float64 `json:"area_h,omitempty"`
	EsPrincipal bool     `json:"es_principal"`
	CreatedAt   Time     `json:"created_at"`
}
